### PR TITLE
[Feature] Added color to accounts labels in transactions list Behind IvyFeatureToggle

### LINKED ISSUE #3545: [FEATURE] Add color to accounts labels in transactions list.
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

Would it be possible for that "label" to appear in the color of the account instead of black... Like it happens with the categories in all transactions list.

### Why do you need it?

It would provide more clarity to know to which account each record belongs with a quick query to the transaction list.

image

### How do you imagine it?

It would be similar to how the categories label is shown in the transaction list but for the accounts label, showing the color that these have defined in their setting in the accounts section.

### PR DESCRIPTION
**Pull request by shamim-emon**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## Screen recording of your changes (if applicable):

https://github.com/user-attachments/assets/820cfa47-0353-45bc-85e3-50ea37b6e199

## What's changed?

Describe with a few bullets **what's new:**
- Added optional functionality to show account colors in transaction

## Risk factors

**What may go wrong if we merge your PR?**
-  N/A

**In what cases won't your code work?**
-  N/A

## Does this PR close any GitHub issues? (do not delete)

- Closes #3545 

## Troubleshooting GitHub Actions (CI) failures ❌
Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt [hunk 1]
```diff
@@ -346,9 +343,15 @@ private fun TransactionHeaderRow(
                 accounts = accounts
             )
 
+            val accountBackgroundColor = if (shouldShowAccountSpecificColorInTransactions) {
+                account?.color?.toComposeColor()
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt:347]: Add the `?: UI.colors.pure` here to simplify the code a bit

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 2]
```diff
@@ -67,6 +67,13 @@ class IvyFeatures @Inject constructor() : Features {
         defaultValue = false
     )
 
+    override val showAccountColorsInTransactions = BoolFeature(
+        key = "show_account_color",
+        group = FeatureGroup.Other,
+        name = "Show Account Color",
+        description = "Show account specific colors for accounts in transactions "
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:74]: Let's be explicit and add `defaultValue = false`

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Overall LGTM! Suggested some minor changes

File: temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt (hunk 1 above)
Comment (shamim-emon) [reply temp/legacy-code/src/main/java/com/ivy/legacy/ui/component/transaction/TransactionCard.kt]: Done

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt (hunk 2 above)
Comment (shamim-emon) [reply shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt]: Done

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**